drivers/sensor/qdec_stm32/qdec_stm32.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2022, Valerio Setti <vsetti@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_qdec
+
+/** @file
+ * @brief STM32 family Quadrature Decoder (QDEC) driver.
+ */
+
+#include <errno.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <soc.h>
```

Comment:
why is `<soc.h>` needed?

Reply:
sorry, that was a leftover from the reference driver I started from